### PR TITLE
Add serve command line options to list all supported model-ids

### PR DESCRIPTION
**Pull request by yutianchen666**

### CODE REVIEW

**Comment by xwu99** [comment]
Could you add a test for this in the pytest?

**Comment by yutianchen666** [comment]
> Could you add a test for this in the pytest?

add new test in test_serve.py

File: llm_on_ray/inference/serve.py [hunk 1]
```diff
@@ -130,6 +139,13 @@ def main(argv=None):
 
     args = parser.parse_args(argv)
 
+    all_models_name = list(all_models.keys())
+    if args.list_model_ids:
+        for model in all_models_name:
+            print(model)
+            print("config_file_path:" + "llm_on_ray/inference/models/" + model + ".yaml")
```
Comment (xwu99) [inline llm_on_ray/inference/serve.py:146]: Could you print model id and config file path in the same line?  you can use a f-string. 
```suggestion
            print(f"{model}: \tllm_on_ray/inference/models/{model}.yaml")
```
and the model yaml filename is not always equal to the model-id right? You need to fetch the real path of the file?

File: tests/inference/test_serve.py [hunk 2]
```diff
@@ -20,20 +20,22 @@
 
 # Parametrize the test function with different combinations of parameters
 @pytest.mark.parametrize(
-    "config_file, models, port, simple, keep_serve_termimal",
+    "config_file, models, port, simple, keep_serve_termimal,list_model_ids",
```
Comment (xwu99) [inline tests/inference/test_serve.py:23]: pay attention to the style
```suggestion
    "config_file, models, port, simple, keep_serve_termimal, list_model_ids",
```

File: tests/inference/test_serve.py [hunk 3]
```diff
@@ -42,25 +44,35 @@ def test_script(
...
+        if simple:
+            cmd_serve.append("--simple")
+        if keep_serve_termimal:
+            cmd_serve.append("--keep_serve_termimal")
 
+    print(cmd_serve)
     result_serve = subprocess.run(cmd_serve, capture_output=True, text=True)
+    if list_model_ids:
+        # Check if the model IDs are listed
+        assert "gpt2" in result_serve.stdout, print("\n" + "Model is not support " + "\n")
```
Comment (xwu99) [inline tests/inference/test_serve.py:71]: why we need to check gpt2 if only model ids are listed?

File: llm_on_ray/inference/serve.py (hunk 1 above)
Comment (yutianchen666) [reply llm_on_ray/inference/serve.py:146]: Okay, i try for if it is more appropriate to define a map

**Review by xwu99 — APPROVED** [review]

**Pull request merged**